### PR TITLE
Doner user register

### PR DESCRIPTION
**Pull request by frederikfink**

### CODE REVIEW

File: src/ts/utils/createUser.ts [hunk 1]
```diff
@@ -6,16 +6,29 @@ import { DonorModel, DonorModelData } from "src/ts/models/DonorModel";
...
-export function createUser(userData: UserModelData | ProducerModelData) {
-    if (userData.userRole === UserTypes.PRODUCER) {
-        let data = <ProducerModelData> userData;
-        return new ProducerModel(data);
-    }
-    else {
-        return new ReceiverModel(userData);
-    }
+
+// export function createUser(userData: UserModelData | DonorModelData | ProducerModelData) {
```
Comment (ChristofferTofteng) [inline src/ts/utils/createUser.ts:9]: Why keep this?

**Review by adamnegaard — APPROVED** [review]
Looks good

File: src/ts/utils/createUser.ts (hunk 1 above)
Comment (frederikfink) [reply src/ts/utils/createUser.ts:9]: added some comments to why to keep this

**Review by ChristofferTofteng — APPROVED** [review]
Seems good, nice work!

File: src/ts/components/elements/Application/Application.tsx [hunk 2]
```diff
@@ -1506,7 +1506,10 @@ class UnwrappedApplication extends React.PureComponent<ApplicationProps, Applica
     }
 
     private pollopolloDonation = async () => {
-        this.setState({ showPollopolloDonation: true });
+        if(this.props.userType === UserTypes.DONOR) this.setState({ showPollopolloDonation: true });
+        else {
+            routes.registerProducer.path;
```
Comment (nimrossum) [inline src/ts/components/elements/Application/Application.tsx:1510]: What should this line do? It currently does nothing

Comment (frederikfink) [reply src/ts/components/elements/Application/Application.tsx:1510]: woops, fixed that on the develop branch 😅

**Pull request merged**